Assets/Scripts/InventoryUI.cs

```diff
@@ -51,6 +54,18 @@ private void ClearAllSlots()
 
     private void SelectItemAtIndex(int index)
     {
-        SelectedItem = slots[index].ItemName; ;
+        SelectedItem = slots[index].ItemName;
+        if (_inventoryAnimation.isMagnifierSelected)
+        {
+            zoomPanel.transform.GetChild(0).gameObject.SetActive(true);
+            var sprite = slots[index].transform.GetChild(1).GetComponent<Image>().sprite;
+            zoomPanel.transform.GetChild(0).transform.GetChild(1).GetComponent<Image>().sprite = sprite;
```

Comment:
this is too complicated, please add suitable variables instead of too much dots in your code